### ISSUE TITLE
[Probably BREAKING] Fix to the usage of the correct scaling arrays at various places

### ISSUE THREAD

**Issue by Ceyron** [Ceyron/exponax#34]
Most importantly for the initial conditions

**Comment by Ceyron** [comment]
Relevant for:
1. Gaussian Random Field Initial Condition
2. Truncated Fourier Series Initial Condition
3. Vorticity Convection with energy injection (alias Kolmogorov Flow)

**Comment by Ceyron** [comment]
Below is an assessment of the repository created in assistance with Claude Code:

### `build_scaling_array`: incorrect mode in `get_spectrum` and IC generators (2D+)

There are two related bugs in how `build_scaling_array` is used. Both are invisible in 1D but cause factor-of-2 errors in 2D and higher.

---

#### 1. `get_spectrum` with `power=True`: DC and Nyquist bins are halved

In `_spectral.py:914-915`:
```python
state_hat_scaled = state_hat / build_scaling_array(..., mode="reconstruction")
magnitude = 0.5 * jnp.abs(state_hat_scaled) ** 2
```

The `"reconstruction"` scaling correctly doubles intermediate modes on the right-most axis (to account for the missing conjugate in the rfft). After squaring, the factor `0.5` compensates — but only for those intermediate modes. For mode 0 (DC) and the Nyquist mode, the reconstruction scaling does *not* double, so the `0.5` incorrectly halves their energy.

**Numerical evidence (1D, N=16):**

| Signal | Bin | Expected | Got | Ratio |
|---|---|---|---|---|
| `x = 3` (DC) | `spectrum[0]` | `\|c₀\|² = 9.0` | `4.5` | **0.5** |
| `5·cos(3·2π/N·n)` | `spectrum[3]` | `2·\|c₃\|² = 12.5` | `12.5` | 1.0 |
| `2·(-1)ⁿ` (Nyquist) | `spectrum[8]` | `\|c_{N/2}\|² = 4.0` | `2.0` | **0.5** |

**Fix:** Replace the blanket `0.5` with a per-mode weight that is `0.5` only where the rfft has a missing conjugate:

```python
rfft_weight = (
    build_scaling_array(num_spatial_dims, num_points, mode="reconstruction")
    / build_scaling_array(num_spatial_dims, num_points, mode="norm_compensation")
)
magnitude = rfft_weight * jnp.abs(state_hat_scaled) ** 2
```

The ratio `reconstruction / norm_compensation` evaluates to `0.5` for intermediate right-most-axis modes (where `reconstruction` gives `N/2` and `norm_compensation` gives `N`) and to `1.0` everywhere else. The `amplitude` path (`power=False`) is unaffected and already correct.

---

#### 2. IC generators in 2D+: `coef_extraction` should be `reconstruction`

Both `RandomTruncatedFourierSeries` (`_truncated_fourier_series.py:134`) and `GaussianRandomField` (`_gaussian_random_field.py:86`) use:

```python
fourier_noise = fourier_noise * build_scaling_array(..., mode="coef_extraction")
u = ifft(fourier_noise, ...)
```

The `"coef_extraction"` mode doubles along *every* axis for non-zero, non-Nyquist modes — i.e., it divides by `N/2` on the non-right-most axes too. But `jnp.fft.irfftn` only adds conjugate entries along the *right-most* axis. Along other axes, each rfft entry stands on its own. So the extra doubling on non-right-most axes inflates the raw FFT values by 2x, but `irfftn` doesn't add the matching conjugate — the result is that the physical amplitude is halved.

**Numerical evidence (2D, N=16, setting amplitude=1 at a single rfft entry):**

| Mode | `coef_extraction` | `reconstruction` | Expected |
|---|---|---|---|
| `(0, 3)` — right-most axis | 1.0 | 1.0 | 1.0 |
| `(3, 0)` — other axis | **0.5** | 1.0 | 1.0 |
| `(3, 5)` — off-axis | **0.5** | 1.0 | 1.0 |

In 1D, the two modes produce identical scaling arrays (there is no "other" axis), so the fix is backward-compatible.

**Fix:** Change `mode="coef_extraction"` to `mode="reconstruction"` in both IC generators.

---

#### Not affected

- **`FourierInterpolator`** — already uses `"reconstruction"`, correct.
- **`map_between_resolutions`** — uses `"norm_compensation"` for a symmetric extract/re-embed round-trip, correct.
- **`fourier_aggregator`** — uses `|X|^p / R` (divides *after* exponentiation), Parseval-consistent for `p=2`, correct.
- **`VorticityConvection2dKolmogorov` injection** — uses `"coef_extraction"` at mode `(0, k)`. Since k₁ = 0, both `coef_extraction` and `reconstruction` give the same scaling (`N` on the zero-wavenumber other-axis). No issue, though switching to `"reconstruction"` would be more semantically accurate.
- **`get_fourier_coefficients`** — user-facing utility where the mode is explicitly chosen. The `coef_extraction` default is documented. However, it is worth noting that `coef_extraction` gives the intuitive "physical amplitude" only for 1D signals and for tensor-product cosines in higher dimensions. For a 2D planar wave `A·cos(k₁x₁ + k₂x₂)`, it returns `2A` at mode `(k₁, k₂)`. The `reconstruction` mode returns `A` in that case.

**Comment by Ceyron** [comment]
Part 1, the issue associated with `get_spectrum` is fixed with #73.
